core/src/main/java/tech/tablesaw/api/NumericColumn.java

```diff
@@ -23,7 +23,21 @@
 
     double[] toDoubleArray();
 
-    float getFloat(int index);
+    default int getInt(int index) {
+        throw new UnsupportedOperationException("getInt() method not supported for all data types");
+    }
+
+    default float getFloat(int index) {
+        throw new UnsupportedOperationException("getFloat() method not supported for all data types");
+    }
+
+    default long getLong(int index) {
+        throw new UnsupportedOperationException("getLong() method not supported for all data types");
+    }
+
+    default double getDouble(int index) {
+        throw new UnsupportedOperationException("getDouble() method not supported for all data types");
+    }
```

Comment:
getShort()?

Reply:
Ahh.... missed this. Thanks. Will add this once the idea proposed here seems reasonable and aligned with rest of the code.